holmes/plugins/toolsets/coralogix/api.py

```diff
@@ -22,9 +42,123 @@ def execute_query(domain: str, api_key: str, query: dict[str, Any]):
+    (start, end) = process_timestamps_to_rfc3339(
+        start_timestamp=params.get("start"),
+        end_timestamp=params.get("end"),
+        default_time_span_seconds=DEFAULT_TIME_SPAN_SECONDS,
+    )
+
+    query_string = build_query_string(config, params)
+    return {
+        "query": query_string,
+        "metadata": {
+            "tier": tier.value,
+            "syntax": "QUERY_SYNTAX_DATAPRIME",
+            "startDate": start,
+            "endDate": end,
+        },
+    }
+
+
+def query_logs_for_tier(
+    config: CoralogixConfig, params: dict, tier: CoralogixTier
+) -> CoralogixQueryResult:
+    http_status = None
+    try:
+        query = build_query(config, params, tier)
+
+        response = execute_http_query(
+            domain=config.domain,
+            api_key=config.api_key,
+            query=query,
+        )
+
+        # Do not print tags if they are repeating the query
+        namespace_name = params.get("namespace_name", None)
+        pod_name = params.get("pod_name", None)
+        add_namespace_tag = not namespace_name and not pod_name
+        add_pod_tag = not pod_name
+        http_status = response.status_code
+        if http_status == 200:
+            logs = parse_logs(
+                raw_logs=response.text.strip(),
+                add_namespace_tag=add_namespace_tag,
+                add_pod_tag=add_pod_tag,
+            )
+            return CoralogixQueryResult(logs=logs, http_status=http_status, error=None)
+        else:
+            return CoralogixQueryResult(
+                logs=[], http_status=http_status, error=response.text
+            )
+    except Exception as e:
+        logging.error("Failed to fetch coralogix logs", exc_info=True)
+        return CoralogixQueryResult(logs=[], http_status=http_status, error=str(e))
+
+
+def query_logs_for_all_tiers(
+    config: CoralogixConfig, params: dict
+) -> CoralogixQueryResult:
+    methodology = config.logs_retrieval_methodology
+    result: CoralogixQueryResult
+
+    if methodology in [
+        CoralogixLogsMethodology.FREQUENT_SEARCH_ONLY,
+        CoralogixLogsMethodology.BOTH_FREQUENT_SEARCH_AND_ARCHIVE,
+        CoralogixLogsMethodology.ARCHIVE_FALLBACK,
+    ]:
+        result = query_logs_for_tier(
+            config=config, params=params, tier=CoralogixTier.FREQUENT_SEARCH
+        )
+
+        if (
+            methodology == CoralogixLogsMethodology.ARCHIVE_FALLBACK and not result.logs
+        ) or methodology == CoralogixLogsMethodology.BOTH_FREQUENT_SEARCH_AND_ARCHIVE:
+            archive_search_results = query_logs_for_tier(
+                config=config, params=params, tier=CoralogixTier.ARCHIVE
+            )
+            result = merge_log_results(result, archive_search_results)
+
+    else:
+        # methodology in [CoralogixLogsMethodology.ARCHIVE_ONLY, CoralogixLogsMethodology.FREQUENT_SEARCH_FALLBACK]:
+        result = query_logs_for_tier(
```

Comment:
Isn't the archive logs already contains the real time logs?

Reply:
Not necessarily